File: backend/src/main/java/com/core/service/UserService.java
package com.core.service;

import com.core.dto.request.CreateUserRequest;
import com.core.dto.request.UpdateUserRequest;
import com.core.dto.response.PageResponse;
import com.core.dto.response.UserResponse;

public interface UserService {

    PageResponse<UserResponse> getUsers(String keyword, String role, Boolean active, int page, int size);

    UserResponse getUserById(Long id);

    UserResponse createUser(CreateUserRequest request);

    UserResponse updateUser(Long id, UpdateUserRequest request);

    void deleteUser(Long id);
}

File: frontend/src/features/users/userService.ts
import axiosInstance from '../../services/axiosInstance';
import type {
  CreateUserRequest,
  UpdateUserRequest,
  UserFilterParams,
  PageResponse,
  UserInfo,
  BaseResponse,
} from './userTypes';

const USERS_URL = '/users';

export const userService = {
  getUsers: (params: UserFilterParams) =>
    axiosInstance.get<BaseResponse<PageResponse<UserInfo>>>(USERS_URL, { params }),

  getUserById: (id: number) =>
    axiosInstance.get<BaseResponse<UserInfo>>(`${USERS_URL}/${id}`),

  createUser: (data: CreateUserRequest) =>
    axiosInstance.post<BaseResponse<UserInfo>>(USERS_URL, data),

  updateUser: (id: number, data: UpdateUserRequest) =>
    axiosInstance.put<BaseResponse<UserInfo>>(`${USERS_URL}/${id}`, data),

  deleteUser: (id: number) =>
    axiosInstance.delete<BaseResponse<void>>(`${USERS_URL}/${id}`),
};
